### ISSUE TITLE
[iOS] Compile error when using delegated property with `@Composable` getters in `object`s

### ISSUE THREAD

**Issue by JunkFood02** [JetBrains/compose-multiplatform#3216]
**Affected platforms**
Select one of the platforms below:
- iOS

**Versions**
Using the [Compose for iOS template](https://github.com/JetBrains/compose-multiplatform-ios-android-template/#readme)

**To Reproduce**
Sample code:
```kotlin
object Surfaces {
    val surfaceDim: Color
        @Composable
        get() = Color.Transparent
}
```

**Error report**

```
> Task :shared:linkPodDebugFrameworkIosSimulatorArm64 FAILED
warning: Cannot infer a bundle ID from packages of source files and exported dependencies, use the bundle name instead: shared. Please specify the bundle ID explicitly using the -Xbinary=bundleId=<id> compiler flag.
error: There is still an unbound symbol after generation of IR module <shared>:
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: theme/Surfaces.surfaceDim.<get-surfaceDim>|-917462939767848401[0]

This could happen if there are two libraries, where one library was compiled against the different version of the other library than the one currently used in the project. Please check that the project configuration is correct and has consistent versions of dependencies.
error: Compilation finished with errors
```

**Additional context**

This only happens with getter annotated with `@Composable` in `object` or `companion object` of classes. Declare such functions at the top level of file won't cause this issue.

**Comment by eymar** [comment]
This was fixed in compose multiplatform compiler plugin 1.4.7-rc01. It will be included automatically in the next compose multiplatform release.

It requires kotlin 1.8.21.

For now it can be enabled like this:

```kotlin
compose {
    kotlinCompilerPlugin.set("org.jetbrains.compose.compiler:compiler:1.4.7-rc01")
   // kotlinCompilerPluginArgs.add("suppressKotlinVersionCompatibilityCheck=1.8.20") // add this only if you want to use kotlin 1.8.20 instead of 1.8.21
}
```

**Comment by jershell** [comment]
i think i have a simular problem.
```
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: androidx.compose.runtime.saveable/SaveableStateHolder.SaveableStateProvider|453623736503830011[0]
```
I used compiler:1.4.7-rc01 & kotlin 1.8.21 but i have same error.

**Comment by eymar** [comment]
@jershell the problem you're describing is a bit different (it's not about a getter).
It should be fixed in 1.4.1 (not released yet). 
___
Does your class need to be public? One of the workarounds is to make it internal if applicable in your case.

**Comment by jershell** [comment]
@eymar I never call this class. I think some dependency call this class inside. I dont now how to search what a class do it.

**Comment by xiaobailong24** [comment]
workaround 
```
object Surfaces {
    @Composable
    fun surfaceDim(): Color = Color.Transparent
}

**Comment by eymar** [comment]
Hey everyone!
The bugs were fixed. Please update to 1.4.1 Compose Multiplatform, and use kotlin 1.8.21 with 

```kotlin
compose {
    kotlinCompilerPlugin.set("1.4.7")
}
```

In the next release, setting `kotlinCompilerPlugin.set("1.4.7")` won't be necessary.

**Comment by igordmn** [comment]
Fixed in 1.4.3

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3216">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
